bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/CommunicationManager.java

```diff
@@ -253,51 +293,63 @@ private void receiveCommand(ItemCommandEvent commandEvent) {
             return link.getItemName().equals(itemName);
         }).filter(link -> {
             // make sure the command event is not sent back to its source
-            return !link.getLinkedUID().toString().equals(commandEvent.getSource());
+            return !link.getLinkedUID().toString().equals(source);
         }).forEach(link -> {
             ChannelUID channelUID = link.getLinkedUID();
             Thing thing = getThing(channelUID.getThingUID());
             if (thing != null) {
-                ThingHandler handler = thing.getHandler();
-                if (handler != null) {
-                    Profile profile = getProfile(link, item, thing);
-                    if (profile instanceof StateProfile) {
-                        safeCaller.create(((StateProfile) profile), StateProfile.class).withAsync()
-                                .withIdentifier(thing).withTimeout(THINGHANDLER_EVENT_TIMEOUT).build()
-                                .onCommandFromItem(command);
+                Channel channel = thing.getChannel(channelUID.getId());
+                if (channel != null) {
+                    @Nullable
```

Comment:
Do we need this annotation? Shouldn't a local variable's nullness state be identified by the respective return value?